buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java

```diff
@@ -145,6 +146,11 @@ private void addNode(String nodeName) {
         this.nodes.add(newNode);
     }
 
+    // This method is called by plugins to override default Url
+    public static void setDistributionUrl(String distributionUrl) {
+        DistributionDownloadPlugin.setDistributionUrl(distributionUrl);
+    }
+
```

Comment:
1. Instead of exposing a public static method, let the interface define this method `TestClusterConfiguration`, `OpenSearchCluster.java` and `OpenSearchNode.java` can override it. 

2. And since its really a custom url, I would rename it to `setCustomDistributionUrl`
3. Instead of directly exposing OpenSearchCluster to DistributionDownloadPlugin, pass this information to OpenSearchNode.java which should set the right distribution. OpenSearchNode.java is designed to setup up the node for tests.
```suggestion
    // This method sets a custom distribution url overriding Ivy repository configured.
    @Override
    public void setCustomDistributionUrl(String customDistributionUrl) {
        ...
    }

```

https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/java/org/opensearch/gradle/testclusters/TestClusterConfiguration.java#L53